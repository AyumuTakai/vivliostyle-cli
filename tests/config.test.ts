import shelljs from 'shelljs';
import path from 'upath';
import { getMergedConfig, maskConfig } from './commandUtil';

const configFiles = [
  'valid.1',
  'valid.2',
  'valid.3',
  'invalid.1',
  'invalid.2',
] as const;
const configFilePath = configFiles.reduce(
  (p, v) => ({
    ...p,
    [v]: path.resolve(__dirname, `fixtures/config/vivliostyle.config.${v}.js`),
  }),
  {} as { [k in typeof configFiles[number]]: string },
);

afterAll(() => {
  shelljs.rm('-f', path.resolve(__dirname, 'fixtures/config/.vs-*'));
});

it('parse vivliostyle config', async () => {
  const validConfig1 = await getMergedConfig(['-c', configFilePath['valid.1']]);
  maskConfig(validConfig1);
  expect(validConfig1).toMatchSnapshot('valid.1.config.js');

  const validConfig2 = await getMergedConfig(['-c', configFilePath['valid.2']]);
  maskConfig(validConfig2);
  expect(validConfig2).toMatchSnapshot('valid.2.config.js');

  const validConfig3 = await getMergedConfig(['-c', configFilePath['valid.3']]);
  maskConfig(validConfig3);
  expect(validConfig3).toMatchSnapshot('valid.3.config.js');
});

it('override option by CLI command', async () => {
  const config = await getMergedConfig([
    '-c',
    configFilePath['valid.1'],
    '-o',
    'yuno.pdf',
    '-o',
    'yuno',
    '-f',
    'webpub',
    '-T',
    'https://myTheme.example.com',
    '-s',
    'JIS-B5',
    '--title',
    'myTitle',
    '--author',
    'myAuthor',
    '--language',
    'myLanguage',
    '--timeout',
    '42',
    '--executable-chromium',
    'myChromium',
    '--style',
    'https://vivlostyle.org',
    '--user-style',
    './user/style/dummy.css',
  ]);
  maskConfig(config);
  expect(config).toMatchSnapshot('valid.1.config.js');
});

it('deny invalid config', () => {
  expect(
    getMergedConfig(['-c', configFilePath['invalid.1']]),
  ).rejects.toThrow();
});

it('deny config which has no entry', () => {
  expect(
    getMergedConfig(['-c', configFilePath['invalid.2']]),
  ).rejects.toThrow();
});

it('deny if any config file or input file is not set', () => {
  expect(getMergedConfig([])).rejects.toThrow();
});

it('Loads same config file on each way', async () => {
  const config1 = await getMergedConfig(['-c', configFilePath['valid.1']]);
  maskConfig(config1);
  const config2 = await getMergedConfig([configFilePath['valid.1']]);
  maskConfig(config2);
  expect(config1).toEqual(config2);
});

it('yields a config with single markdown', async () => {
  const config = await getMergedConfig([
    path.resolve(__dirname, 'fixtures/config/sample.md'),
  ]);
  maskConfig(config);
  expect(config.entries[0].target).toMatch(
    /^__WORKSPACE__\/tests\/fixtures\/config\/\.vs-.+\.sample\.html$/,
  );
  expect(config.manifestPath).toMatch(
    /^__WORKSPACE__\/tests\/fixtures\/config\/\.vs-.+\.publication\.json$/,
  );
  const entryAlias = config.exportAliases.find(
    ({ source }) => source === config.entries[0].target,
  );
  expect(entryAlias?.target).toMatch(
    '__WORKSPACE__/tests/fixtures/config/sample.html',
  );
  const manifestAlias = config.exportAliases.find(
    ({ source }) => source === config.manifestPath,
  );
  expect(manifestAlias?.target).toMatch(
    '__WORKSPACE__/tests/fixtures/config/publication.json',
  );
  config.manifestPath = '__SNIP__';
  config.entries[0].target = '__SNIP__';
  (config.exportAliases as unknown) = '__SNIP__';
  expect(config).toMatchSnapshot();
});

it('imports single html file', async () => {
  const config = await getMergedConfig([
    path.resolve(__dirname, 'fixtures/config/sample.html'),
  ]);
  maskConfig(config);
  expect(config).toMatchSnapshot();
});

it('yields a config with single input and vivliostyle config', async () => {
  const config = await getMergedConfig([
    path.resolve(__dirname, 'fixtures/config/sample.md'),
    '-c',
    configFilePath['valid.1'],
  ]);
  maskConfig(config);
  expect(config.entries[0].target).toMatch(
    /^__WORKSPACE__\/tests\/fixtures\/config\/\.vs-.+\.sample\.html$/,
  );
  expect(config.manifestPath).toMatch(
    /^__WORKSPACE__\/tests\/fixtures\/config\/\.vs-.+\.publication\.json$/,
  );
  config.manifestPath = '__SNIP__';
  config.entries[0].target = '__SNIP__';
  (config.exportAliases as unknown) = '__SNIP__';
  expect(config).toMatchSnapshot();
});

it('imports a EPUB file', async () => {
  const config = await getMergedConfig([
    path.resolve(__dirname, 'fixtures/epubs/adaptive.epub'),
    '-o',
    'epub.pdf',
  ]);
  maskConfig(config);
  expect(config.epubOpfPath).toMatch(/OPS\/content\.opf$/);
  config.epubOpfPath = '__SNIP__';
  expect(config).toMatchSnapshot();
});

it('imports a EPUB OPF file', async () => {
  const config = await getMergedConfig([
    path.resolve(__dirname, 'fixtures/epubs/adaptive/OPS/content.opf'),
    '-o',
    'epub-opf.pdf',
  ]);
  maskConfig(config);
  expect(config).toMatchSnapshot();
});

it('imports a webbook compliant to W3C Web publication', async () => {
  const config = await getMergedConfig([
    path.resolve(__dirname, 'fixtures/webbooks/w3c-webpub/publication.json'),
    '-o',
    'w3c-webpub',
  ]);
  maskConfig(config);
  expect(config).toMatchSnapshot();
});

it('imports a webbook compliant to Readium Web publication', async () => {
  const config = await getMergedConfig([
    path.resolve(__dirname, 'fixtures/webbooks/readium-webpub/manifest.jsonld'),
    '-o',
    'readium-webpub',
  ]);
  maskConfig(config);
  expect(config).toMatchSnapshot();
});

it('imports a https URL', async () => {
  const config = await getMergedConfig([
    'https://vivliostyle.github.io/vivliostyle_doc/ja/vivliostyle-user-group-vol1/',
  ]);
  maskConfig(config);
  expect(config).toMatchSnapshot();
});
