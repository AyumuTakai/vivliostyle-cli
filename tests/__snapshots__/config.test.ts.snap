// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports a EPUB OPF file 1`] = `
Object {
  "cover": undefined,
  "entries": Array [],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/epubs/adaptive/OPS",
  "epubOpfPath": "__WORKSPACE__/tests/fixtures/epubs/adaptive/OPS/content.opf",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/epubs/adaptive/OPS/content.opf",
    "format": "epub-opf",
  },
  "language": null,
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/epub-opf.pdf",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/epubs/adaptive/OPS",
}
`;

exports[`imports a EPUB file 1`] = `
Object {
  "cover": undefined,
  "entries": Array [],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/epubs",
  "epubOpfPath": "__SNIP__",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/epubs/adaptive.epub",
    "format": "epub",
  },
  "language": null,
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/epub.pdf",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/epubs",
}
`;

exports[`imports a https URL 1`] = `
Object {
  "cover": undefined,
  "entries": Array [],
  "entryContextDir": "__WORKSPACE__",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "https://vivliostyle.github.io/vivliostyle_doc/ja/vivliostyle-user-group-vol1/",
    "format": "webbook",
  },
  "language": null,
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/output.pdf",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "webbookEntryPath": "https://vivliostyle.github.io/vivliostyle_doc/ja/vivliostyle-user-group-vol1/",
  "workspaceDir": "__WORKSPACE__",
}
`;

exports[`imports a webbook compliant to Readium Web publication 1`] = `
Object {
  "cover": undefined,
  "entries": Array [],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/webbooks/readium-webpub",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/webbooks/readium-webpub/manifest.jsonld",
    "format": "pub-manifest",
  },
  "language": null,
  "manifestAutoGenerate": null,
  "manifestPath": "__WORKSPACE__/tests/fixtures/webbooks/readium-webpub/manifest.jsonld",
  "outputs": Array [
    Object {
      "format": "webpub",
      "path": "__WORKSPACE__/readium-webpub",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/webbooks/readium-webpub",
}
`;

exports[`imports a webbook compliant to W3C Web publication 1`] = `
Object {
  "cover": undefined,
  "entries": Array [],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/webbooks/w3c-webpub",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/webbooks/w3c-webpub/publication.json",
    "format": "pub-manifest",
  },
  "language": null,
  "manifestAutoGenerate": null,
  "manifestPath": "__WORKSPACE__/tests/fixtures/webbooks/w3c-webpub/publication.json",
  "outputs": Array [
    Object {
      "format": "webpub",
      "path": "__WORKSPACE__/w3c-webpub",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/webbooks/w3c-webpub",
}
`;

exports[`imports single html file 1`] = `
Object {
  "cover": undefined,
  "entries": Array [],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/config",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/config/sample.html",
    "format": "webbook",
  },
  "language": null,
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/output.pdf",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "webbookEntryPath": "__WORKSPACE__/tests/fixtures/config/sample.html",
  "workspaceDir": "__WORKSPACE__/tests/fixtures/config",
}
`;

exports[`override option by CLI command: valid.1.config.js 1`] = `
Object {
  "cover": "__WORKSPACE__/tests/fixtures/config/cover.png",
  "entries": Array [
    Object {
      "rel": "contents",
      "target": "__WORKSPACE__/tests/fixtures/config/workspaceDir/toc.html",
      "theme": Object {
        "location": "https://myTheme.example.com",
        "name": "myTheme.example.com",
        "type": "uri",
        "vars": undefined,
      },
      "title": "Table of Contents",
    },
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/manuscript.md",
      "target": "__WORKSPACE__/tests/fixtures/config/workspaceDir/manuscript.html",
      "theme": Object {
        "location": "https://myTheme.example.com",
        "name": "myTheme.example.com",
        "type": "uri",
        "vars": undefined,
      },
      "title": "myTitle",
      "type": "text/markdown",
    },
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/manuscript.md",
      "target": "__WORKSPACE__/tests/fixtures/config/workspaceDir/manuscript.html",
      "theme": Object {
        "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/theme.css",
        "location": "__WORKSPACE__/tests/fixtures/config/theme.css",
        "name": "theme.css",
        "type": "file",
        "vars": undefined,
      },
      "title": "title",
      "type": "text/markdown",
    },
  ],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/config",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/config/workspaceDir/publication.json",
    "format": "pub-manifest",
  },
  "language": "myLanguage",
  "manifestAutoGenerate": Object {
    "author": "myAuthor",
    "title": "myTitle",
  },
  "manifestPath": "__WORKSPACE__/tests/fixtures/config/workspaceDir/publication.json",
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/yuno.pdf",
    },
    Object {
      "format": "webpub",
      "path": "__WORKSPACE__/yuno",
    },
  ],
  "pressReady": true,
  "sandbox": true,
  "size": Object {
    "format": "JIS-B5",
  },
  "themeIndexes": Array [
    Object {
      "location": "https://myTheme.example.com",
      "name": "myTheme.example.com",
      "type": "uri",
      "vars": undefined,
    },
    Object {
      "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/theme.css",
      "location": "__WORKSPACE__/tests/fixtures/config/theme.css",
      "name": "theme.css",
      "type": "file",
      "vars": undefined,
    },
  ],
  "themeVars": undefined,
  "timeout": 42000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/config/workspaceDir",
}
`;

exports[`parse vivliostyle config: valid.1.config.js 1`] = `
Object {
  "cover": "__WORKSPACE__/tests/fixtures/config/cover.png",
  "entries": Array [
    Object {
      "rel": "contents",
      "target": "__WORKSPACE__/tests/fixtures/config/workspaceDir/toc.html",
      "theme": Object {
        "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/themes/packages/debug-theme",
        "location": "__WORKSPACE__/tests/fixtures/themes/debug-theme",
        "name": "debug-theme",
        "style": "./theme.css",
        "type": "package",
        "vars": undefined,
      },
      "title": "Table of Contents",
    },
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/manuscript.md",
      "target": "__WORKSPACE__/tests/fixtures/config/workspaceDir/manuscript.html",
      "theme": Object {
        "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/themes/packages/debug-theme",
        "location": "__WORKSPACE__/tests/fixtures/themes/debug-theme",
        "name": "debug-theme",
        "style": "./theme.css",
        "type": "package",
        "vars": undefined,
      },
      "title": "title",
      "type": "text/markdown",
    },
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/manuscript.md",
      "target": "__WORKSPACE__/tests/fixtures/config/workspaceDir/manuscript.html",
      "theme": Object {
        "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/theme.css",
        "location": "__WORKSPACE__/tests/fixtures/config/theme.css",
        "name": "theme.css",
        "type": "file",
        "vars": undefined,
      },
      "title": "title",
      "type": "text/markdown",
    },
  ],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/config",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/config/workspaceDir/publication.json",
    "format": "pub-manifest",
  },
  "language": "language",
  "manifestAutoGenerate": Object {
    "author": "author",
    "title": "title",
  },
  "manifestPath": "__WORKSPACE__/tests/fixtures/config/workspaceDir/publication.json",
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/fixtures/config/output1.pdf",
    },
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/fixtures/config/output2.pdf",
    },
  ],
  "pressReady": true,
  "sandbox": true,
  "size": Object {
    "format": "size",
  },
  "themeIndexes": Array [
    Object {
      "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/themes/packages/debug-theme",
      "location": "__WORKSPACE__/tests/fixtures/themes/debug-theme",
      "name": "debug-theme",
      "style": "./theme.css",
      "type": "package",
      "vars": undefined,
    },
    Object {
      "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/theme.css",
      "location": "__WORKSPACE__/tests/fixtures/config/theme.css",
      "name": "theme.css",
      "type": "file",
      "vars": undefined,
    },
  ],
  "themeVars": undefined,
  "timeout": 1,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/config/workspaceDir",
}
`;

exports[`parse vivliostyle config: valid.2.config.js 1`] = `
Object {
  "cover": undefined,
  "entries": Array [
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/manuscript.md",
      "target": "__WORKSPACE__/tests/fixtures/config/manuscript.html",
      "theme": undefined,
      "title": "pkgName",
      "type": "text/markdown",
    },
  ],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/config",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/config/publication.json",
    "format": "pub-manifest",
  },
  "language": null,
  "manifestAutoGenerate": Object {
    "author": "pkgAuthor",
    "title": "pkgName",
  },
  "manifestPath": "__WORKSPACE__/tests/fixtures/config/publication.json",
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/fixtures/config/output1.pdf",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/config",
}
`;

exports[`parse vivliostyle config: valid.3.config.js 1`] = `
Object {
  "cover": undefined,
  "entries": Array [
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/manuscript.md",
      "target": "__WORKSPACE__/tests/fixtures/config/manuscript.html",
      "theme": undefined,
      "title": "example",
      "type": "text/markdown",
    },
  ],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/config",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": Array [],
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/config/publication.json",
    "format": "pub-manifest",
  },
  "language": null,
  "manifestAutoGenerate": Object {
    "author": "pkgAuthor",
    "title": "example",
  },
  "manifestPath": "__WORKSPACE__/tests/fixtures/config/publication.json",
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/fixtures/config/example.pdf",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/config",
}
`;

exports[`yields a config with single input and vivliostyle config 1`] = `
Object {
  "cover": "__WORKSPACE__/tests/fixtures/config/cover.png",
  "entries": Array [
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/sample.md",
      "target": "__SNIP__",
      "theme": Object {
        "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/themes/packages/debug-theme",
        "location": "__WORKSPACE__/tests/fixtures/themes/debug-theme",
        "name": "debug-theme",
        "style": "./theme.css",
        "type": "package",
        "vars": undefined,
      },
      "title": "Yuno",
      "type": "text/markdown",
    },
  ],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/config",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": "__SNIP__",
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/config/sample.md",
    "format": "markdown",
  },
  "language": "language",
  "manifestAutoGenerate": Object {
    "author": "author",
    "title": "title",
  },
  "manifestPath": "__SNIP__",
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/fixtures/config/output1.pdf",
    },
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/fixtures/config/output2.pdf",
    },
  ],
  "pressReady": true,
  "sandbox": true,
  "size": Object {
    "format": "size",
  },
  "themeIndexes": Array [
    Object {
      "destination": "__WORKSPACE__/tests/fixtures/config/workspaceDir/themes/packages/debug-theme",
      "location": "__WORKSPACE__/tests/fixtures/themes/debug-theme",
      "name": "debug-theme",
      "style": "./theme.css",
      "type": "package",
      "vars": undefined,
    },
  ],
  "themeVars": undefined,
  "timeout": 1,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/config/workspaceDir",
}
`;

exports[`yields a config with single markdown 1`] = `
Object {
  "cover": undefined,
  "entries": Array [
    Object {
      "source": "__WORKSPACE__/tests/fixtures/config/sample.md",
      "target": "__SNIP__",
      "theme": undefined,
      "title": "Yuno",
      "type": "text/markdown",
    },
  ],
  "entryContextDir": "__WORKSPACE__/tests/fixtures/config",
  "executableChromium": "__EXECUTABLE_CHROMIUM_PATH__",
  "exportAliases": "__SNIP__",
  "includeAssets": Array [
    "**/*.png",
    "**/*.jpg",
    "**/*.jpeg",
    "**/*.svg",
    "**/*.gif",
    "**/*.webp",
    "**/*.apng",
    "**/*.ttf",
    "**/*.otf",
    "**/*.woff",
    "**/*.woff2",
  ],
  "input": Object {
    "entry": "__WORKSPACE__/tests/fixtures/config/sample.md",
    "format": "markdown",
  },
  "language": null,
  "manifestAutoGenerate": Object {
    "author": "",
    "title": "Yuno",
  },
  "manifestPath": "__SNIP__",
  "outputs": Array [
    Object {
      "format": "pdf",
      "path": "__WORKSPACE__/tests/output.pdf",
    },
  ],
  "pressReady": false,
  "sandbox": true,
  "size": undefined,
  "themeIndexes": Array [],
  "themeVars": undefined,
  "timeout": 120000,
  "verbose": false,
  "workspaceDir": "__WORKSPACE__/tests/fixtures/config",
}
`;
